app/controllers/api/v1/rooms_controller.rb

```diff
@@ -51,13 +51,13 @@ def public_show
       def create
         # TODO: amir - ensure accessibility for authenticated requests only.
         # The created room will be the current user's unless a user_id param is provided with the request.
-        # user_id = room_params[:user_id]
-        room = Room.create(name: room_params[:name], user_id: room_params[:user_id])
+        @new_room = Room.create(name: room_params[:name], user_id: room_params[:user_id])
 
         return render_error errors: user.errors.to_a if hcaptcha_enabled? && !verify_hcaptcha(response: params[:token])
 
-        if room.save
-          logger.info "room(friendly_id):#{room.friendly_id} created for user(id):#{room.user_id}"
+        if @new_room.save
+          generate_access_code
```

Comment:
Wouldn't it be better to just create a `room` and pass in that room to the `generate_access_code` function instead of creating a global var `new_room` ?

Reply:
Yes, passing the room as an argument would be better.
I can't recall why I did it that way.
This whole PR is getting a big revamp soon - the logic is being put inside the Room model instead of in the controller.